crates/cli/src/meta/types/authoring/v2.rs

```diff
@@ -185,6 +200,55 @@ impl AuthoringMetaV2 {
 
         Ok(meta)
     }
+
+    /// checks if the given contract implements ERC165
+    /// the process is done as described per ERC165 specs:
+    /// https://eips.ethereum.org/EIPS/eip-165#how-to-detect-if-a-contract-implements-erc-165
+    pub async fn supports_erc165(client: &ReadableClientHttp, contract_address: Address) -> bool {
+        let parameters = ReadContractParameters {
+            address: contract_address,
+            // equates to 0x01ffc9a701ffc9a700000000000000000000000000000000000000000000000000000000
+            call: IERC165::supportsInterfaceCall {
+                interfaceID: IERC165::supportsInterfaceCall::SELECTOR.into(),
+            },
+            block_number: None,
+        };
+        let result = client.read(parameters).await.map(|v| v._0).unwrap_or(false);
+        if !result {
+            return false;
+        }
+
+        let parameters = ReadContractParameters {
+            address: contract_address,
+            // equates to 0x01ffc9a7ffffffff00000000000000000000000000000000000000000000000000000000
+            call: IERC165::supportsInterfaceCall {
+                interfaceID: [255, 255, 255, 255].into(),
+            },
+            block_number: None,
+        };
+        let result = client.read(parameters).await.map(|v| v._0).unwrap_or(true);
+        !result
+    }
+
+    /// checks if the given contract implements IDescribeByMetaV1 interface
+    pub async fn implements_i_describe_by_meta_v2(
```

Comment:
this is more general than just AuthoringMetaV2, so this doesn't feel like the right place for this to live either

Reply:
relocated to ./src/meta/mod.rs